packages/dashboard/app/withdraw/utils/handlers.ts

```diff
@@ -0,0 +1,245 @@
+  unhideToken: (address: string) => void,
+  setSelectedToken: (token: TokenBalance | null) => void,
+  setWithdrawAmount: (amount: string) => void
+) => {
+  e.stopPropagation();
+
+  if (hiddenTokens.includes(address.toLowerCase())) {
+    unhideToken(address);
+  } else {
+    hideToken(address);
+
+    // If the hidden token was selected, clear selection
+    if (selectedToken && selectedToken.address.toLowerCase() === address.toLowerCase()) {
+      setSelectedToken(null);
+      setWithdrawAmount('');
+    }
+  }
+};
+
+export const handleRefreshBalances = async (
+  ethAddress: string | undefined,
+  setLoading: (loading: boolean) => void,
+  setBalances: (balances: TokenBalance[]) => void,
+  showStatus: ShowStatusFn,
+  showError: ShowErrorFn
+) => {
+  if (!ethAddress) return;
+
+  try {
+    setLoading(true);
+    const ethBalance = await fetchEthBalance(ethAddress);
+    const erc20TokenBalances = await fetchERC20TokenBalances(ethAddress);
+
+    const balances = [ethBalance.balances[0],...erc20TokenBalances.balances];
+
+    if (erc20TokenBalances.success && ethBalance.success) {
+      setBalances(balances);
+      showStatus('Token balances fetched successfully', 'success');
+    } else {
+      showStatus('Failed to fetch all token balances', 'warning');
+      if (erc20TokenBalances.error) {
+        showError(erc20TokenBalances.error, 'Token Balance Error');
+      }
+      if (ethBalance.error) {
+        showError(ethBalance.error, 'Token Balance Error');
+      }
+    }
+  } catch (err: any) {
+    showStatus('Failed to fetch token balances', 'error');
+    showError('Error fetching token balances', 'Error');
+  } finally {
+    setLoading(false);
+  }
+};
+
+export const handleSubmit = async (
+  e: React.FormEvent,
+  selectedToken: TokenBalance | null,
+  withdrawAmount: string,
+  withdrawAddress: string,
+  agentPKP: IRelayPKP,
+  sessionSigs: SessionSigs,
+  setSubmitting: (submitting: boolean) => void,
+  setWithdrawAmount: (amount: string) => void,
+  setWithdrawAddress: (address: string) => void,
+  showStatus: ShowStatusFn,
+  showError: ShowErrorFn,
+  refreshBalances: () => void
+) => {
+  e.preventDefault();
+
+  if (!selectedToken || !withdrawAmount || !withdrawAddress) {
+    showStatus('Please fill all fields', 'warning');
+    return;
+  }
+
+  if (parseFloat(withdrawAmount) === 0) {
+    showStatus('Withdrawal amount cannot be zero', 'warning');
+    return;
+  }
+
+  if (!ethers.utils.isAddress(withdrawAddress)) {
+    showStatus('Invalid withdrawal address', 'error');
+    return;
+  }
+
+  if (selectedToken) {
+    if (selectedToken.symbol === 'ETH') {
+      const parsedAmount = ethers.utils.parseUnits(withdrawAmount, selectedToken.decimals);
+      if (parsedAmount.gte(selectedToken.rawBalance)) {
+        showStatus(`You have entered an amount greater than or equal to your balance. Please enter a smaller amount.`, 'warning');
+        return;
+      }
+    }
+  }
```

Comment:
This should also check for the required gas. The check should be the same as in other parts of code

nit: `selectedToken` is guaranteed to have a value at this point. First `if` is pointless